src/vs/workbench/api/common/extHostTypes.ts

```diff
@@ -2915,6 +2915,46 @@ export class DebugAdapterInlineImplementation implements vscode.DebugAdapterInli
 	}
 }
 
+
+@es5ClassCompat
+export class StackFrameFocus {
+	static isStackFrameFocus(thing: any): thing is StackFrameFocus {
+		if (thing instanceof StackFrameFocus) {
+			return true;
+		}
+		if (!thing) {
+			return false;
+		}
+
+		return !!(<StackFrameFocus>thing).session;
+	}
+
+	constructor(
+		public readonly session: vscode.DebugSession,
+		readonly threadId?: number,
+		readonly frameId?: number) { }
+}
+
+@es5ClassCompat
+export class ThreadFocus {
+	static isThreadFocus(thing: any): thing is StackFrameFocus {
+		if (thing instanceof ThreadFocus) {
+			return true;
+		}
+		if (!thing) {
+			return false;
+		}
+
+		return !!(<StackFrameFocus>thing).session && !('frameId' in thing);
```

Comment:
for session, this just checks whether its set; wasn't sure how to handle the check, since this is defined in /browser, which cannot (rather, should not) be referenced from this source.